contracts/mainnet/connectors/uniswapV3/main.sol

```diff
@@ -0,0 +1,211 @@
+pragma solidity ^0.7.6;
+pragma abicoder v2;
+
+/**
+ * @title Uniswap v3.
+ * @dev Decentralized Exchange.
+ */
+
+import {TokenInterface} from "../../common/interfaces.sol";
+import {Helpers} from "./helpers.sol";
+import {Events} from "./events.sol";
+
+abstract contract UniswapResolver is Helpers, Events {
+    uint256 private _lastMintIndex = 1;
```

Comment:
@cryptoDev222 We can't have a storage variable. We need to fetch it from nft manager contract.

Reply:
```solidity
function getNftLastId(address user) public view returns(uint256 id) {
        uint256 len = nonfungiblePositionManager.balanceOf(user);
        id = nonfungiblePositionManager.tokenOfOwnerByIndex(user, len - 1);
    }
```

Reply:
Okay, got it, Thank you.